qemu/tests/ple_test.py

```diff
@@ -0,0 +1,77 @@
+import re
+import logging
+
+from avocado.utils import process
+from virttest import env_process
+from virttest import error_context
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    ept test:
+    1) Check if ple is enabled on host, if not, enable it
+    2) Boot guest and run unixbench inside guest
+    3) Record benchmark scores and shutdown guest
+    4) Disable ple on host
+    5) Boot guest and run unixbench inside guest again
+    6) Compare benchmark scores with step 3)
+    7) Restore env, enable ple
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    def set_ple(value):
+        """
+        Set moudle kvm_intel/kvm_amd ple parameter
+        """
+        process.system(unload_cmd)
+        process.system(load_cmd % value)
+
+    def run_unixbench(cmd):
+        """
+        Run unixbench inside guest, return benchmark scores
+        """
+        error_context.context("Run unixbench inside guest", logging.info)
+        output = session.cmd_output_safe(cmd, timeout=4800)
+        scores = re.findall(r"System Benchmarks Index Score\s+(\d+\.?\d+)",
+                            output)
+        return [float(i) for i in scores]
+
+    error_context.context("Enable ple on host if it's disabled", logging.info)
+    unload_cmd = params["unload_cmd"]
+    load_cmd = params["load_cmd"]
+    ple_value = process.getoutput(params["read_cmd"])
+    if ple_value == 0:
+        set_ple(params["ple_value"])
+
+    host_cpu_count = process.getoutput(params["get_cpu_cmd"], shell=True)
+    params["smp"] = host_cpu_count
```

Comment:
Not must, framework can calculate it automatically, use `vcpu_maxcpus` for this case.